gpslogger/src/main/java/com/mendhak/gpslogger/loggers/csv/CSVFileLogger.java

```diff
@@ -46,34 +47,22 @@ public CSVFileLogger(File file, @Nullable Float batteryLevel) {
 
     @Override
     public void write(Location loc) throws Exception {
-        if (!file.exists()) {
-            file.createNewFile();
-
-            FileOutputStream writer = new FileOutputStream(file, true);
-            BufferedOutputStream output = new BufferedOutputStream(writer);
-            String header = "time,lat,lon,elevation,accuracy,bearing,speed,satellites,provider,hdop,vdop,pdop,geoidheight,ageofdgpsdata,dgpsid,activity,battery\n";
-            output.write(header.getBytes());
-            output.flush();
-            output.close();
-
+        if (!Session.getInstance().hasDescription()) {
```

Comment:
I think this is needed, because `write` always gets called, while `annotate` only when a description is present. I based it on the CustomUrlLogger, which does it that way. Else I think the events with annotations are logged twice.

https://github.com/mendhak/gpslogger/blob/master/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/customurl/CustomUrlLogger.java#L52